records/2023-11-21.quicksight.md

```diff
@@ -0,0 +1,56 @@
+# QuickSight
+
+Date: 2023-11-21
+
+## Status
+
+**DRAFT**.
+
+## Context
+
+To have a thorough understanding of data in our system, the Notify team wanted a collaborative tool that would allow them to explore the database. While we already have installed Blazer, we need a tool with more business intelligence capabilities that is more accessible to non-developers. AWS QuickSight was chosen to be our short/mid term solution for a BI tool that has no sensitive data, that external CDS teams (ie other than Notify) can tap into, and which is easily queryable and fast.
+
+## Implementation
+
+Introducing QuickSight into our system requires us to do the following:
+
+1. QuickSight needs access to the main Notify DB
+2. QuickSight needs to be secure and auditable
+
+## Deployment
+
+### QuickSight
+
+QuickSight is an AWS product that we can run in ca-central-1. We can use terraform to create and modify AWS Quicksight subscriptions in our accounts.
+
+### Data
+
+QuickSight imports data from the Notify database into optimized data storage called SPICE. This allows queries to run faster and without creating a load on RDS. SPICE datasets (ie tables) are updated either manually or on schedule. Updates can either be full or incremental.
+
+We specify the data fields added to the QuickSight datasets. We are excluding any fields containing PII or user data (for example, email addresses, phone numbers, and template personalisation).
+
+QuickSight is outside our VPC and requires a AWS QuickSight VPC connection to access our database.
+
+### User access
+
+QuickSight can be configured so that AWS SSO is needed to access the QuickSight subscription. To access the data sets users must also be added one of the QuickSight user groups `dataset-owners` or `dataset-viewers`. Owners may change datasets or add new datasets through the AWS Console (though we will likely do this through terraform). Viewers can only view datasets and use them to create analyses.
+
+### Auditing
+
+QuickSight can be configured to log audit data to CloudTrail.
```

Comment:
So without PII this is less critical but it should be enabled in case PII slips through, and just as a general CYA for us.

Reply:
👍 As we discovered this is enabled by default. 🎉